src/network/serverpackethandler.cpp

```diff
@@ -1101,8 +1101,15 @@ void Server::handleCommand_Damage(NetworkPacket* pkt)
 				<< (int)damage << " hp at " << PP(player->getPosition() / BS)
 				<< std::endl;
 
-		playersao->setHP(playersao->getHP() - damage);
-		SendPlayerHPOrDie(playersao);
+		u8 oldhp = playersao->getHP();
+		playersao->setHP(oldhp - damage);
+
+		// Make sure that the player only dies once.
+		if (oldhp - playersao->getHP() > 0 && player->isDead()) {
```

Comment:
Errm no you do the math wrongly @SmallJoker  its equal to `if (damage < 0 && player->hp == 0) {`. Minus and minus gives plus :)